### DISCUSSION TITLE
Replacing Beta nuget dependencies

### DISCUSSION THREAD

**Discussion by razvangoga** [tye-v2/tye2#37] [Ideas]
At the moment tye2 relies on 3 beta and unmaintained nuget packages
- https://www.nuget.org/packages/System.CommandLine (unmaintained since jun 2022)
- https://www.nuget.org/packages/System.CommandLine.rendering (unmaintained since jun 2022)
- https://www.nuget.org/packages/Bedrock.Framework (unmaintained since mar 2022)

We should take a look at each one, see what they do and replace them with something more standard and maintained

<img width="554" alt="image" src="https://github.com/tye-v2/tye2/assets/1067494/ddfabbf3-9313-4cba-bc1f-09756aeabba7">

cc @drakon660

**Comment by razvangoga** [comment]
Thread for 
- https://www.nuget.org/packages/System.CommandLine (unmaintained since jun 2022)
- https://www.nuget.org/packages/System.CommandLine.rendering (unmaintained since jun 2022)

**Reply by Mrxx99** [reply]
There were no new nuget packages for some time but the repo does not look unmaintained (last commit 3 weeks ago)

**Reply by razvangoga** [reply]
may be, but without a new public release, it's useless for others - i'm not gonna build the nuget for it myself...

**Reply by razvangoga** [reply]
i mean, why would we when there a mature oss solutions like this : https://spectreconsole.net/cli/ we could use (and maybe others)

**Reply by drakon660** [reply]
@Mrxx99 if you have other proposal, maybe you used some different one.

**Reply by Mrxx99** [reply]
I have only used https://github.com/Tyrrrz/CliFx and I like it, but Spectre.Console I have also looked at and also looks like a great choice (I thought about switching to it in a project). Didn't want to argue against replacing System.CommandLine just wanted to point out that it is not dead.

**Comment by razvangoga** [comment]
Thread for 
- https://www.nuget.org/packages/Bedrock.Framework (unmaintained since mar 2022)
- https://github.com/davidfowl/BedrockFramework

**Reply by drakon660** [reply]
#38

**Reply by drakon660** [reply]
Lets leave that for now.

**Comment by drakon660** [comment]
<img width="623" alt="image" src="https://github.com/tye-v2/tye2/assets/3002778/eb5f7af3-a90f-4c2f-ab78-9547c72e061f">

One simple PR to update Microsoft.Extensions.Configuration and Serilog.Sinks.Seq, and next one with open tracing?

**Comment by drakon660** [comment]
Thread for 

- https://www.nuget.org/packages/OpenTelemetry/1.7.0

I have started to investigating it, there are of course breaking changes, I went through even few blogs regarding tracing topic, and examples from those blogs didn't work also with current version, but blogs were from last year, so there is a definitely a problem with using open tracing lib in .NET, breaking changes  occur really frequently.  I am gonna practice more with open tracing and check how to rewrite code to work with current version.

#45 Please check this one.

After merging this #45 I will go through OpenTelemetry update.
